src/rpc/subscription.cpp

```diff
@@ -0,0 +1,61 @@
+#include "subscription.h"
+#include "rpc.grpc.pb.h"
+#include "rpc.pb.h"
+#include "rpc_tools.h"
+#include "transaction.h"
+#include <memory>
+#include <string>
+
+Subscriber::Subscriber(std::unique_ptr<rpc::Subscription::Stub>&& stub, uint8_t service_)
+    : service(service_), push_stub_(std::move(stub)) {}
+
+bool Subscriber::PushBlock(rpc::Vertex& vertex) {
+    rpc::EmptyResponse response;
+    grpc::ClientContext context;
+    return push_stub_->PushBlock(&context, vertex, &response).ok();
+}
+
+
+bool Subscriber::PushTx(rpc::Transaction& tx) {
+    rpc::EmptyResponse response;
+    grpc::ClientContext context;
+    return push_stub_->PushTx(&context, tx, &response).ok();
+}
+
+void Publisher::AddNewSubscriber(std::string address, uint8_t service) {
+    auto stub =
+        std::make_unique<rpc::Subscription::Stub>(grpc::CreateChannel(address, grpc::InsecureChannelCredentials()));
+    if (stub) {
+        subscribers_.insert_or_assign(address, Subscriber{std::move(stub), service});
+    }
+}
+
+void Publisher::DeleteSubscriber(std::string address) {
+    subscribers_.erase(address);
+}
+
+void Publisher::PushMsg(void* msg, SubType type) {
+    switch (type) {
+        case SubType::BLOCK: {
+            rpc::Vertex vtx = *ToRPCVertex(*(Vertex*) msg);
```

Comment:
I think resources here are not deleted. Refer to rpc_tool.cpp. The same for line 49 `rpc::transaction`